src/main/java/io/numaproj/numaflow/sideinput/Message.java

```diff
@@ -0,0 +1,35 @@
+package io.numaproj.numaflow.sideinput;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * Message is used to wrap the data returned by Side Input Retriever.
+ */
+
+@Getter
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class Message {
+    private final byte[] value;
+    private final boolean noBroadcast;
+
+    /**
+     * BroadcastMessage creates a new Message with the given value
+     * This is used to broadcast the message to other side input vertices.
+     * @param value message value
+     * @return returns the Message with noBroadcast flag set to false
+     */
+    public static Message broadcastMessage(byte[] value) {
```

Comment:
naming - `broadcastMessage` gives an impression that the method is doing the broadcasting work while it just creates an object. Consider renaming to `createBroadcastMessage`. Same for `noBroadcastMessage`.